src/v/cluster/topic_table.cc

```diff
@@ -277,6 +277,36 @@ topic_table::apply(update_topic_properties_cmd cmd, model::offset o) {
     co_return make_error_code(errc::success);
 }
 
+ss::future<std::error_code>
+topic_table::apply(create_materialized_topic_cmd cmd, model::offset o) {
+    const model::topic_namespace& source = cmd.value;
+    const model::topic_namespace& new_materialized_topic = cmd.key;
+    if (_topics.contains(new_materialized_topic)) {
+        co_return make_error_code(errc::topic_already_exists);
+    }
```

Comment:
what will happen if source topic will be deleted ?

Reply:
Good point. To handle that is additional logic that i didn't want to add in this PR, so there's another PR that depends on this one that explicitly supports deleting materialized logs and/or source topics: https://github.com/graphcareful/redpanda/pull/14